app/src/main/java/samyups/example/trainingss/MainActivity.kt

```diff
@@ -1,11 +1,28 @@
 package samyups.example.trainingss
 
-import android.support.v7.app.AppCompatActivity
+import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
+
+    val dogAdapter = DogAdapter()
+    private var layoutManager: RecyclerView.LayoutManager? = null
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        initDogRV()
+    }
+
+    private fun initDogRV() {
+        dogRV.apply {
```

Comment:
Interesting, I see you didn't have to do a `findViewById` here. I guess this is a new feature because back in the day we had to do `findViewById`.  I wonder...